language-support/ts/daml-ledger/index.ts

```diff
@@ -566,6 +591,7 @@ class Ledger {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const off = (type: string, listener: any): void => void emitter.off(type, listener);
     const close = (): void => {
+      emitter.emit('close', {code: 1000, reason: 'Stream.close() was called'});
```

Comment:
This may be worth exploring as a separate PR. The notion of `close` on `Stream` seems a bit confused at the moment. As the code stands (without this PR), if you register a listener for `'close'` events, you get notified when the underlying WebSocket closes (even though the stream will then try to open a new one and, if it manages to reconnect, will keep sending `'change'` events), and you do not get notified when the stream itself gets closed (by calling `.close()` on it), because the code in `.close()` removes the listeners before closing the WebSocket connection (and trying to send a message to `'close'` listeners). Further, calling `.close()` also does not really close the stream: while it does remove all the listeners and close the WebSocket connection, it also immediately creates a new WebSocket connection to replace the closed one, as it does not remove the `onClose` listener on the WebSocket itself.

I believe the following would be a better interface, but am a bit unsure about how to change this without breaking backwards compatibility:
- WebSocket connection closing is an internal detail and should not leak out (assuming we do manage to reconnect).
- Listeners for `'close'` events should trigger when the stream will no longer emit events: either `.close()` has been called, or the WebSocket connection has failed beyond the code's willingness to try reconnecting.
- When `.close()` is called, the WebSocket connection should be properly terminated (i.e. no reconnection).
- Calling `.close()` on an already closed stream can be defined as either an error or a no-op, depending on how forgiving we want to be (I'd prefer an error).

As of this PR, `'close'` listeners get called when `.close()` is called, but none of the other issues are addressed. Consequently, the `'close'` event is still in the WebSocket `'close'` event format. I chose the "normal termination" code of 1000 as it seemed semantically correct here (described as "the connection has reached its natural end after fulfilling its purpose").

Reply:
I have removed this change and will open a separate PR for this. This on is big enough as it stands.